### ISSUE TITLE
Better cauldrons featuring better custom blocks!

### ISSUE THREAD

**Issue by Kierancraft** [WolfyScript/CustomCrafting-Wiki#64]
**Is your feature request related to a problem? Please describe.**
It's not related to a problem.

**Describe the solution you'd like**
 There should be a cauldron revamp that allows for any block to be selected and the replacement block (e.g. composter level 7 replaced with level 6 or wheat replaced with beetroot when a red dye is thrown at it) and multiple items allowed (like floor crafting from old datapacks) and then particles. It would allow for no blocks to be selected or even blacklisting blocks (such as making it never work in a fence post or wall). Then an advanced option for relative block positions (relative to other blocks or in specific directions, like north or east must be glass / place TNT somewhere 3 blocks away and make a honey block exactly opposite, also on-top of a waxed lightly weathered cut copper stairs)

**Describe alternatives you've considered**
Floor crafting option in crafting menu

**Additional context**
Great plugin, many bugs, overall 8.5/10
Would be better if it had this or something like it.
Also, just thought id add, the custom items section should have a way of altering block break time with each tool/item and a way of altering the block drops so my custom steel ore doesn't drop iron. (Maybe even altering the drops when a custom tool is used?)

**Comment by Kierancraft** [comment]
all of the examples are purely hypothetical by the way and in no way influenced by my server, i picked random blocks from memory

**Comment by WolfyScript** [comment]
So a crafting system like it's used by old command block creations? 
Like, you drop items onto the ground, and they form an item?

I don't like the idea of ground crafting. Even back when I created command block contraptions, I used droppers for custom recipes instead, because there are too many complications. 
For example, a player just drops items on the ground without wanting to craft them, how should the plugin know? 
There needs to be a direct interaction of the player to initiate crafting.... dropping items on the ground is such an essential part of the game that you cannot hijack.

In that case, I would treat it as a separate feature, neither as part of the cauldron nor without a required workstation.
Perhaps the composter can have its own type of recipes, that activate when you right-click on them.

Regarding your idea that there could be structure requirements, that could be part of crafting recipes or even a condition for all kinds of recipes.

**Comment by Kierancraft** [comment]
Yes, a data pack is a command block creation except in a file and with loot tables and, well, custom crafting table recipes, but that was only added in 1.14 I think so yes, old command block creations. The main thing I want to do on my server is have a player throw a bucket into a composted when it’s raining and 30sec later get a filled bucket. Also, I think the bug I was getting the other day was from a placed custom item. Does the custom item system work with server restarts? Also I want to make an item on my server called a hammer that breaks stone into cobblestone (like a pick axe) and cobble into gravel, then dirt, then sand, then a custom item called dust. I also want them to be axes and I need them to break stone and cobblestone faster. Is there anyway I can do this? Thanks for reading my paragraph.

TL;DR: Hammers break stone>cobble>gravel>dirt>sand>custom item dust. Hammers are axes and can’t break stone and cobble fast. They also don’t have the drops. Any functionality in future?
